packages/nx/src/command-line/release/release.ts

```diff
@@ -139,9 +139,15 @@ export async function release(
       });
     }
   }
+  const hasNewVersion =
```

Comment:
`workspaceVersion === null; // workspace version is applicable no changes, no new version required`

`workspaceVersion === undefined; // a unified version for the workspace is not applicable (independent projects) but it says nothing about whether or not a new version is required`

```suggestion
  let hasNewVersion = false;
  // null means that all projects are versioned together but there were no changes
  if (workspaceVersion !== null) {
    hasNewVersion = Object.values(versionResult.projectsVersionData).some(
      (version) => version.newVersion !== null
    );
  }

```